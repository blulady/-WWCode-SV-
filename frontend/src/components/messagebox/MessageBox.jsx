import React from "react";

import styles from "./MessageBox.module.css";
import cx from "classnames";

const MessageBox = (props) => {
    const type = {
        Info: "Info",
        Success: "Success",
        Error: "Error"
    };

    const getTypeStr = (str) => {
       return type[str] || this.type.Info;
    };

    return (<div className={cx(styles[getTypeStr(props.type).toLowerCase()], styles["message-box"], "d-flex align-items-center justify-content-center flex-column")}>
        <div className={styles["title"]}>{props.title}</div>
        <div className={styles["message"]}>
        {props.message}
        </div>
    </div>);
};
export default MessageBox;