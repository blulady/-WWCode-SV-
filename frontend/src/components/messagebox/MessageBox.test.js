import React from "react";
import { render } from "@testing-library/react";
import MessageBox from "./MessageBox";

describe("MessageBox", () => {
  test("it renders without crashing", () => {
    const { container } = render(
      <MessageBox
        type="Error"
        title="Sorry!"
        message="We could not process your request. \nPlease try again."
      />
    );
    expect(container).toMatchSnapshot();
  });

  test("it renders correct styling for error", () => {
    const { container } = render(
      <MessageBox
        type="Error"
        title="Sorry!"
        message="We could not process your request. \nPlease try again."
      />
    );
    const typeDiv = container.querySelector(".message-box");

    expect(typeDiv).toHaveClass("error");
  });

  test("it renders correct styling for success", () => {
    const { container } = render(
      <MessageBox
        type="Success"
        title="Success!"
        message="Request Completed."
      />
    );
    const typeDiv = container.querySelector(".message-box");

    expect(typeDiv).toHaveClass("success");
  });
});
