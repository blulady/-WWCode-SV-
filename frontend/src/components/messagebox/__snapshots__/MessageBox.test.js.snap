// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox it renders without crashing 1`] = `
<div>
  <div
    class="error message-box d-flex align-items-center justify-content-center flex-column"
    data-testid="message-box"
  >
    <div
      class="title"
    >
      Sorry!
    </div>
    <div
      class="message"
    >
      We could not process your request. \\nPlease try again.
    </div>
  </div>
</div>
`;
